### ISSUE TITLE
Not enough disk space for user check os disk quota settings

### ISSUE THREAD

**Issue by IngoVals** [ValveSoftware/steam-for-linux#7731]
#### Your system information

* Steam client version (build number or date): Mar 23 2021
* Distribution (e.g. Ubuntu): Linux Mint
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:

When trying to auto update a game on my secondary library folder ( at /steam ) steam complains about disk space.
**not enough disk space for user check os disk quota settings**
There is plenty of space, settings for library folders also shows plenty of free space.
I''ve not used a quota for any user. I tried to look into that and seems it is set at 0 meaning no limit. I've even tried changing it to a large limit with no success. I'm assuming that the active user is used, or is there a specific STEAM user?

```
     Filesystem   space   quota   limit   grace   files   quota   limit   grace
      /dev/sda5  60080M    103G    118G           37428       0       0        
```

**Comment by score100** [comment]
I have the same issue here :(

**Comment by IngoVals** [comment]
Mine seems to have resolved, but no idea how.

One theory I had was that I actually needed space for the full install instead of just the update. So if you have a 30gb game with 5gb update you would need 30gb of free space instead of just the 5gb.

**Comment by jamzy118** [comment]
I see this when trying to install 5 GB update on a HDD having 120 GB free. it's arch linux, no quota enabled.

**Comment by Saroumane** [comment]
I just reproduced the issue while trying to re-install "ARK: Survival Evolved" on an empty 3TB HDD. (the steamapps/common folder is on this drive). I don't use quota.
The game package to download is about 43 GB. 
The game itself needs about 130 GB of disk when installed.

My `steamapps/downloading` and `steamapp/common` folders are symlinked on another, smaller SSD drive with *only* 100 GB free space.


So for my case: maybe Steam clients erroneously checks for 130GB free space on the drive that holds `steamapps/temp` or `steamapps/downloading` folders, instead of checking the drive that holds the destination folder ( `steampapps/common` )

**Comment by testnetprotonmail** [comment]
sudo mv ~/.local/share/Steam /second-disk somehow solved it for me, don't know why but it downloads normally now

**Comment by af7567** [comment]
I was getting this error today when trying to download a 13.4GB update for Stalker 2 on a disk with 90GB free space. It seems that the 13.4GB is misleading and the update actually wants 141GB which can be seen when hovering the mouse over the error message (which is also misleading since most of us don't use quotas on our home PC :) ).

After reading the comments above and symlinking my `steamapps/downloading` directory to one on a bigger disk the problem seems to be fixed.

**Comment by JJones780** [comment]
Thanks @af7567 for noticing the mouse hover number!!
### The upgrade requires the amount indicated when you hover the mouse over the error message.  The symlink trick is a great trick if you can't free up enough space on the current drive. 

My experience ( the long story):

- Misleadingly stated it needed 4.1GB for upgrade, hover over error said 40GB.
- I freed up 60GB and was able to do the update:
- It filled up the steam/steamapps/downloading/### dir with 37.7GB ( only 4.1GB of which was from downloading, the rest I assume was **copied** from the existing install)
- When it finished the above download dir went to 0 and the steam/steamapps/common/NameOfGame increased in size by 2.8GB (from  82.4 to 85.2GB).
- The overall /home/me dir got up to aprox 779GB  and back down to 744.1GB after the upgrade.  This checked with the numbers earlier ( ie 34.9GB + 2.8GB = 37.7GB ).
- So in summary.  It required 37.7GB of space and freed 34.9GB when it was done leaving only 2.8GB of new storage required.  Also 4.1-2.8 = 1.3GB of changed data.

**Comment by lonkalsim** [comment]
this is still a bug, right? my SSD is pretty small, I have Overwatch to update (Steam says 4.1GB but when hovering into the error I see 53.29GB), but I don't have this much space and I definitely have required files, so I'm forced to reinstall the whole game because of it.

**Comment by popcar2** [comment]
Weirdly enough I also found this thread after trying to update Overwatch, it might be a game-specific issue because I had over 150GB of free storage and Steam was convinced I don't have enough space to update it...

**Comment by DedAzaMarks** [comment]
Still have this issue. I want to update 007 First Light. Patch 1.2.1 itself weights 16.8Gbs. According to df i have 29Gb of free space. Steam refers to disk quota setting, but i don't have anything like that.
```
❯ df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/nvme0n1p7  249G  209G   29G  89% /
```

to resolve this, i deleted another space-heavy game, but steam download manager showed that exactly 16.8Gbs were downloaded

**Comment by JJones780** [comment]
> to resolve this, i deleted another space-heavy game, but steam download manager showed that exactly 16.8Gbs were downloaded


**But did you check the number shown when you hovered the mouse over the error?**  
It's not the patch size... it's the resulting patched file size.  It's either patching a copy or more likely the patching process needs to have the original file available as it grows the patched file.

I just checked to see if maybe they had added an option to do "risky" in-place patching... but no.

@popcar2 is that 150GB free on the same drive as the game install?  

It's probably not a bug... other than Valve needs to better lay out the numbers including the total space required to pull off the patching. It would save a lot of frustration!
